expression/expression.go

```diff
@@ -28,6 +31,10 @@ func (err ErrInvalidExpression) Error() string {
 	return err.internalError.Error()
 }
 
+type verificationResult struct {
```

Comment:
Why do you need a new structure? There is `ErrrInvalidExpression` above and, as I understood, `verificationResult` can contain an error only if the expression is invalid, so `ErrrInvalidExpression` fits your needs quite well. But maybe I'm misunderstanding something, can you please explain it

Reply:
I will remove the caching of verificationResult in the next commit, therefore there will be no need for this struct.